bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/services/impl/DynamicDeckServiceImpl.java

```diff
@@ -0,0 +1,224 @@
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(assetResourceIterator, Spliterator.ORDERED),
+                false).collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    @Override
+    public List<Resource> fetchAssetListFromTags(String tagsString, ResourceResolver resourceResolver) {
+        TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+        List<Resource> arrayList = new ArrayList<>();
+        String[] tagsArray = StringUtils.split(tagsString, DynamicDeckDynamoConstants.COMMA);
+        Arrays.stream(tagsArray).forEach(tag -> {
+            RangeIterator<Resource> tags = tagManager.find(tag);
+            tags.forEachRemaining(taggedResource -> {
+                if (taggedResource.getPath().startsWith(DynamicDeckDynamoConstants.DAM_ROOT) && taggedResource.getPath().endsWith(DynamicDeckDynamoConstants.SLASH + DynamicDeckDynamoConstants.DAM_METADATA)) {
+                    Asset taggedAsset = taggedResource.getParent().getParent().adaptTo(Asset.class);
+                    if (null != taggedAsset) {
+                        arrayList.add(taggedAsset.adaptTo(Resource.class));
+                    }
+                }
+            });
+        });
+        return arrayList;
+    }
+
+    private void addExportJobProperty(Asset assetObject, Object exportJobId) {
+        if (assetObject == null) {
+            LOGGER.error("Asset is null, hence export properties cannot be added.");
+            return;
+        }
+        Resource assetResource = assetObject.adaptTo(Resource.class);
+        if (assetResource == null || ResourceUtil.isNonExistingResource(assetResource)) {
+            LOGGER.error("Asset resource is null or doesn't exist, hence export properties cannot be added. {}", assetObject.getPath());
+            return;
+        }
+        Resource content = assetResource.getChild(JcrConstants.JCR_CONTENT);
+        if (content == null || ResourceUtil.isNonExistingResource(content)
+                && content.adaptTo(ModifiableValueMap.class) == null) {
+            LOGGER.error("Asset JCR content is null or doesn't exist, hence export properties cannot be added. {}", assetResource.getPath());
+            return;
+        }
+        ModifiableValueMap contentProperties = content.adaptTo(ModifiableValueMap.class);
+        if (contentProperties == null) {
+            LOGGER.error("Property object is null, hence export properties cannot be added {}", assetResource.getPath());
+            return;
+        }
+        contentProperties.put("exportJobId", exportJobId);
+        contentProperties.put(DamConstants.DAM_ASSET_STATE, DamConstants.DAM_ASSET_STATE_PROCESSING);
+        contentProperties.put(DynamicDeckDynamoConstants.PN_INDD_TEMPLATE_TYPE, DynamicDeckDynamoConstants.DECK_TYPE);
+
+    }
+
+
+    private Job addJob(Asset master, Map<String, Object> props, JobManager jobManager) throws DynamicDeckDynamoException {
```

Comment:
why a ```JobManager``` parameter, if this service holds an @Reference to the Sling JobManager (the variable name is actually the same)?